### PR TITLE
N°4792 - memory limit is reached when an AttributeExternalKey is pointing to a class with an AttributeBlob

### PR DESCRIPTION
**Pull request by eespie**
What was done:
  * When requesting values for external keys in the case of combobox (request all possible values) the columns fetched are limitted to the minimum to avoid loading the full objects including blobs
  * __toString() was also added to DBSearch to ease debug

### CODE REVIEW

File: application/ui.extkeywidget.class.inc.php [hunk 1]
```diff
@@ -211,14 +211,23 @@ public function DisplaySelect(WebPage $oPage, $iMaxComboLength, $bAllowTargetCre
 			$sClassAllowed = $oAllowedValues->GetClass();
 			$bAddingValue = false;
 
+			// #4792 - load only the required fields
```
Comment (Molkobain) [inline application/ui.extkeywidget.class.inc.php]:
```suggestion
			// N°4792 - load only the required fields
```

Just change this as #XXX are usually used for SourceForge ticket reference.

**Review by Molkobain — CHANGES_REQUESTED** [review]
Except from the "typo" seems all good to me ✌

_Edit: You should be able to apply the suggestion directly from the PR UI_

**Comment by Molkobain** [comment]
_Also, for internal PRs can you add the "internal" label? Thanks_

**Comment by eespie** [comment]
changed comment and fix data used from MetaModel::GetNameSpec()

**Comment by Hipska** [comment]
Problem was the whole object from the external key was loaded? And the solution is now only the fields to make the friendlyname? Great improvement!

One question, will a programmer still be able to get any other field (by lazy loading) like this: `$oObject->Get('external_id')->Get('blob_field')`?

**Comment by eespie** [comment]
The use of OptimizeColumnLoad() method is to be used with caution. If another column is requested with Get() then the object will be fully reloaded. So yes, it works, but the performances will be very bad.
In this specific case, the object is not used anymore after building the list.

**Pull request merged**